### ISSUE TITLE
[ENH] Add wrappers around scikit-learn explained_variance_score and r2_score metrics

### ISSUE THREAD

**Issue by RNKuhns** [alan-turing-institute/sktime#1476]
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is.

`sktime` currently has a variety of forecasting error metrics (where a lower value is better). Adding support for common "score" metrics (where a higher value is better) would be nice to round out our forecasting performance metrics. 

**Describe the solution you'd like**
We can provide a sktime compatiable interface to scikit-learn's implementation of a couple common "score" metrics. The metrics that make the most sense to me for use in evaluating forecasting models are:

- [explained_variance_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.explained_variance_score.html#sklearn.metrics.explained_variance_score)
- [r2_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html#sklearn.metrics.r2_score)

Adding these should be relatively straight-forward. The first step is to implement a function for each in `sktime.performance_metrics.forecasting._functions.py` and the contributor can generally follow along with what is done for `mean_absolute_error` in that file. 

Then there needs to be the class versions of the metrics in implemented in `sktime.performance_metrics.forecasting._classes.py`. Both metrics can inherit from `_BaseForecastingScoreMetric` and just need to define `__init__` appropriately (see `MeanAbsoluteError` as a guide).

**Describe alternatives you've considered**
Primary alternative is not adding the metrics. 

**Additional context**
We need to provide an interface, instead of just using the scikit-learn versions, to ensure that they fit with sktime's forecasting performance metric interface (need to optionally accept y_train without throwing an error).

**Comment by deepto98** [comment]
@RNKuhns Hi, I'd like to work on this issue, can you please assign this to me?

**Comment by RNKuhns** [comment]
@deepto98 consider it yours to work on. Let me know when you've got a PR for me to look at. And feel free to ask me any questions.

**Comment by RNKuhns** [comment]
@deepto98 I wanted to check-in with you to see how it was going. Let me know if you want to talk about the implementation at all.

**Comment by fkiraly** [comment]
@deepto98, is this still ongoing?

**Comment by ka-ori** [comment]
@RNKuhns  @fkiraly  Can you assign this issue to me. I would like to work on it since the previous assignee hasn't. Thank you

**Comment by ka-ori** [comment]
@marrov Can i get review for this pr please?
